main.py

```diff
@@ -742,22 +742,29 @@ def make_move(move, board:Board):
     #update king position
     if board.board[position] == K or board.board[position] == k:
         board.king_position[board.side] = target
-    else:
+    #update other pieces positions
+    elif board.board[position] != K or board.board[position] != k:
         if char_ascii[board.board[position]] == '.': 
             board.undo_move()
             return 0  #the second check. first was in line 730
+        if capture:
+            captured_piece_list_position = get_piece_positions_from_letter(char_ascii[board.board[target]],board)
+            captured_piece_list_position.remove(target)
+
         piece_positions = get_piece_positions_from_letter(char_ascii[board.board[position]],board)
-        print("piece_positions")
-        print(piece_positions)
-        print(position)
+
         piece_positions.remove(position)
         #promoted pawn
         if promoted_piece:
             board.board[target] = promoted_piece
             promotions_positions = get_piece_positions_from_letter(char_ascii[board.board[target]],board)
             promotions_positions.append(target)
-        else:    #regular move
+        elif not promoted_piece:    #regular move
             piece_positions.append(target)
+    
+    #make the move
+    board.board[target] = board.board[position]
+    board.board[position] = e
```

Comment:
the move should be made after all the if conditions have been tested, please move this to the end of make_move

Reply:
it is now almost at the end. it has to be before this if condition
![image](https://github.com/Garic152/pythonchess/assets/85243017/9036fb3f-ae5f-4b01-a646-5cae2548408b)

Reply:
That make sense :+1: